.github/workflows/tests.yml

```diff
@@ -2,7 +2,7 @@ name: Tests
 
 on:
   push:
-    branches: [master]
+    branches: ['*']
```

Comment:
Reverted.

Could you tell me about this decision? I hope to test with branches as I develop with different feature branches.

Reply:
Even having unlimited time for an open source project on GitHub Actions, we have pretty strong limits in performance, which result in only several parallel jobs running per repo (which are also pretty slow).

So, because of these limitations, we need to keep the list of possible runs pretty short:
- push a tag
- push to `master` = merge a PR
- create and update a PR
- scheduled weekly run on the `master` of code analysis (and in my PR build) workflow(s)

If you need to test something on CI, you can create a PR for that branch or temporarily do this change (but don't forget to revert it back when the PR is created)